### ISSUE TITLE
Failed to flush the buffer. Error_class=NoMethodError error="undefined method reason_phrase

### ISSUE THREAD

**Issue by laurentiu92** [uken/fluent-plugin-elasticsearch#443]
#### Problem

The plugin is refusing to connect to elasticsearch cluster. In the logs can be seen that there are multiple warnings and at the end is failing to flush the buffer. 
Important to mention that the plugin has this behavior when is used with the swift plugin. (Swift plugin is working correctly)

#### Steps to replicate

Provide example config and message

```aconf
<source>
  @type secure_forward
  shared_key XXXXXX
  self_hostname "${HOSTNAME}"
  secure false
</source>

 
<match output_tag>
    @type copy
    <store>
     @type elasticsearch_dynamic
     include_tag_key true
     hosts <IP>,<IP>,<IP>
     port 9200
     logstash_format true
     logstash_prefix openshift-noprod-${record['kubernetes']['namespace_name'] || 'unknown'}
     <buffer>
       @type memory
       flush_interval 5s
       queue_limit_length 32
       chunk_limit_size 2M
       retry_forever
       flush_thread_count 8
       retry_max_interval 30
     </buffer>
    </store>
    <store>
     @type relabel
     @label @NOTIFICATION
    </store>
</match>

<label @NOTIFICATION>
      <filter output_tag>
        @type grep
        <regexp>
         key $.kubernetes.container_name
         pattern oma-payments
        </regexp>
      </filter>
      <match output_tag>
         @type swift
        auth_url https://localhost:5000/v2.0/tokens
        auth_user XXXXXXXXXXXX
        auth_api_key XXXXXXXXXXXX
        auth_region RegionOne
        auth_tenant OM_Addons_DevOps
        swift_container om_addons_payments
        swift_object_key_format %{path}%{time_slice}_%{index}.json
        ssl_verify false
        path logs/
        buffer_path /var/log/fluent/swift
        time_slice_format %Y%m%d-%H
        time_slice_wait 10m
        utc
      </match>
</label>
```

#### Expected Behavior or What you need to ask

...

#### Using Fluentd and ES plugin versions

* OS version Centos 7
Fluentd v1.0
 
* ES plugin fluent-plugin-elasticsearch (2.4.0)
  
```log
 2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/faraday-0.13.1/lib/faraday/adapter/excon.rb:55:in `call'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/faraday-0.13.1/lib/faraday/rack_builder.rb:141:in `build_response'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/faraday-0.13.1/lib/faraday/connection.rb:387:in `run_request'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/http/faraday.rb:23:in `block in perform_request'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/base.rb:262:in `perform_request'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/transport/http/faraday.rb:20:in `perform_request'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/elasticsearch-transport-5.0.4/lib/elasticsearch/transport/client.rb:131:in `perform_request'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/elasticsearch-api-5.0.4/lib/elasticsearch/api/actions/ping.rb:20:in `ping'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluent-plugin-elasticsearch-2.4.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:64:in `client'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluent-plugin-elasticsearch-2.4.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:221:in `rescue in send_bulk'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluent-plugin-elasticsearch-2.4.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:216:in `send_bulk'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluent-plugin-elasticsearch-2.4.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:209:in `block in write'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluent-plugin-elasticsearch-2.4.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:208:in `each'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluent-plugin-elasticsearch-2.4.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:208:in `write'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/plugin/output.rb:1093:in `try_flush'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/plugin/output.rb:1318:in `flush_thread_run'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/plugin/output.rb:439:in `block (2 levels) in start'
  2018-07-10 17:16:46 +0200 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/plugin_helper/thread.rb:78:in `block in thread_create'
2018-07-10 17:16:46 +0200 [warn]: #0 failed to flush the buffer. retry_time=10 next_retry_seconds=2018-07-10 17:17:18 +0200 chunk="570a69d52f7ae63996a7b311e35d17b2" error_class=NoMethodError error="undefined method `reason_phrase' for #
<Excon::Response:0x00007fbb40ca3c28>"
```
* ES version (optional)

**Comment by cosmo0920** [comment]
> Important to mention that the plugin has this behavior when is used with the swift plugin. (Swift plugin is working correctly)

This means that ES plugin should work without swift plugin?
Or, ES plugin is always broken under your environment?

**Comment by laurentiu92** [comment]
Yes, ES is working correctly when is not used with swift. 
Use cases:
1. Only ES plugin used: plugin working as expected
2. ES with file plugin using the presented configuration: both plugins working correctly
3. ES with swift plugin using the presented configuration: only swift pligin is working

**Comment by cosmo0920** [comment]
Now, @fujimotos is investigating this issue.

**Comment by fujimotos** [comment]
@cosmo0920 @laurentiu92 I've investigated this issue.

It turned out that the current version of fluent-pluign-swift depends on `fog (~> 1.15.0)`,
which, in turn, depends on `excon (~> 0.25.0`). So if we add fluent-pluign-swift to the
dependency list, we naturally end up using `excon` v0.25.x.

The fundamental problem here is that `excon` v0.25.x is really, really old, and
fluent-plugin-elasticsearch (which also depends on excon) *does* require newer versions
of this library.

For example, let's take this error message in the bug report:

> NoMethodError error="undefined method `reason_phrase' for # <Excon::Response:0x00007fbb40ca3c28>"

This occurs, because the method `Excon::Response.reason_phrase` has been added in
excon v0.40.2, and does not exist in excon 0.25.x.
